modules/js/open-close.js

```diff
@@ -1,535 +1,560 @@
-    deleteConstraint(compositeArray[3].bodies[0], compositeArray[0].bodies[0])
-  }
-  Body.setPosition(compositeArray[0].bodies[0], {x:compositeArray[0].constraints[0].pointA.x, y:compositeArray[0].constraints[0].pointA.y} )
-  Body.setAngle(compositeArray[1].bodies[0], 0)
-  radius = 80;
-  if(crankMod){
-    radius = radius + 52
-  }
-  compositeArray[1].radius = radius;
-  steps = (0.25 * radius)*2;
-  toothWidthDegree = 2;
-  toothWidth = (toothWidthDegree/conversionFactor);
-  if(camMod == true){
-    if(compositeArray[1].shape == "cam"){
-      changeBody2(1)
-    }
-    else if(compositeArray[1].shape == "shell"){
-      changeShell()
-    }
-  }
-  else if(rackPinionMod == true && compositeArray[1].alternate == false){
-    changeBodyContinuous(1)
-  }
-  else if(crankMod){
-    changeBodyCircle(1)
-    createConstraint2(compositeArray[1].bodies[0], compositeArray[0].bodies[0])
-    createConstraintFake(compositeArray[0].bodies[0], compositeArray[2].bodies[0])
-    createConstraintFake(compositeArray[0].bodies[0], compositeArray[3].bodies[0])
-  }
-  else{
-    changeBody(1);
-  }
-  if(rackPinionMod == true){
-    compositeArray[1].constraints[0].pointA.x = (window.innerWidth)*(0.75*0.45)+(radius+(toothHeight*1.8))
-  }
-  createUIConstraints(compositeArray[0], beamSpace, 0,6)
-  pivotHeight(linkageLength)
+  createUIConstraints(compositeArray[0], beamSpace, 0, 6);
+  pivotHeight(linkageLength);
 }
 ///////////////////////////////////////////////////////////////////////////////////////////////
 // reset/initialize rack and pinion motion
-function rackPinion(){
-  updateToothWidth()
-  resetRadius()
+function rackPinion() {
+  updateToothWidth();
+  resetRadius();
   // remove ui body constraints/linkages
-  removeUIConstraints(compositeArray[0])
+  removeUIConstraints(compositeArray[0]);
   // remove all constraints
-  deleteConstraint(compositeArray[1].bodies[0], compositeArray[0].bodies[0])
-  deleteConstraint(compositeArray[2].bodies[0], compositeArray[0].bodies[0])
-  deleteConstraint(compositeArray[3].bodies[0], compositeArray[0].bodies[0])
+  deleteConstraint(compositeArray[1].bodies[0], compositeArray[0].bodies[0]);
+  deleteConstraint(compositeArray[2].bodies[0], compositeArray[0].bodies[0]);
+  deleteConstraint(compositeArray[3].bodies[0], compositeArray[0].bodies[0]);
   // set step size
-  steps = (0.25 * radius)*2;
+  steps = 0.25 * radius * 2;
   // redraw and create gear and lingear
   changeBody4(0);
   changeBody(1);
   // set position of gear and lingear
-  Body.setPosition(compositeArray[1].bodies[0], {x:(window.innerWidth)*(0.75*0.45)+(radius+(toothHeight*1.8)), y:(window.innerHeight)*(0.68) + rackPinBase})
-  Body.setPosition(compositeArray[0].bodies[0], {x:compositeArray[0].bodies[0].position.x, y:(window.innerHeight)*(0.8) + rackPinBase})
+  Body.setPosition(compositeArray[1].bodies[0], {
+    x: window.innerWidth * (0.75 * 0.45) + (radius + toothHeight * 1.8),
+    y: window.innerHeight * 0.68 + rackPinBase,
```

Comment:
All these magic numbers! I guess we are going to see this A LOT.  Maybe a good TODO item would be to refactor some of these as constants in the future.